Src/FluentAssertions/Execution/LateBoundTestFramework.cs

```diff
@@ -1,37 +1,70 @@
 ﻿using System;
 using System.Diagnostics.CodeAnalysis;
+using System.IO;
+using System.Linq.Expressions;
 using System.Reflection;
 
 namespace FluentAssertions.Execution;
 
 internal abstract class LateBoundTestFramework : ITestFramework
 {
-    private Assembly assembly;
+    private readonly bool loadAssembly;
+    private Func<string, Exception> exceptionFactory;
 
-    [DoesNotReturn]
-    public void Throw(string message)
+    protected LateBoundTestFramework(bool loadAssembly = false)
     {
-        Type exceptionType = assembly.GetType(ExceptionFullName);
+        this.loadAssembly = loadAssembly;
+        exceptionFactory = _ => throw new InvalidOperationException($"{nameof(IsAvailable)} must be called first.");
+    }
 
-        if (exceptionType is null)
+    [DoesNotReturn]
+    public void Throw(string message) => throw exceptionFactory(message);
+
+    public bool IsAvailable
+    {
+        get
         {
-            throw new NotSupportedException(
-                $"Failed to create the assertion exception for the current test framework: \"{ExceptionFullName}, {assembly.FullName}\"");
+            var assembly = GetAssembly();
+            var exceptionType = assembly?.GetType(ExceptionFullName);
+            exceptionFactory = exceptionType != null
+                ? GetExceptionFactory(exceptionType)
+                : _ => throw new InvalidOperationException($"{GetType().Name} is not available");
+            return exceptionType is not null;
         }
-
-        throw (Exception)Activator.CreateInstance(exceptionType, message);
     }
 
-    public bool IsAvailable
+    private Assembly GetAssembly()
     {
-        get
+        var assembly = Array.Find(AppDomain.CurrentDomain.GetAssemblies(), a => a.GetName().Name == AssemblyName);
+        if (assembly is null && loadAssembly)
         {
-            string prefix = AssemblyName + ",";
+            try
+            {
+                return Assembly.Load(new AssemblyName(AssemblyName));
+            }
+            catch (FileNotFoundException)
+            {
+                return null;
+            }
```

Comment:
`When_xunit3_is_used_it_should_throw_xunit_exceptions_for_assertion_failures` covers this `catch (FileNotFoundException)` since it tries to load `xunit.assert` first and fails. But adding a `None.Specs` that references none of the supported testing frameworks and asserting that the fallback `AssertionFailedException` is thrown is probably a good idea anyway.

For the second `catch` (around `expression.Compile()`) it's going to be very hard to test. I added it as a safety thinking that maybe `Compile()` might fail on some platforms and falling back on `Activator.CreateInstance` which was the original implementation. But maybe this whole expression thing is over-engineering since we're throwing an exception anyway.

Reply:
> adding a `None.Specs` that references none of the supported testing frameworks

That was a stupid idea! How are the tests going to be run without any testing framework? 😂

Reply:
> When_xunit3_is_used_it_should_throw_xunit_exceptions_for_assertion_failures covers this catch (FileNotFoundException) since it tries to load xunit.assert first and fails.

Coveralls mentioned this as "no coverage": see https://coveralls.io/builds/69359638/source?filename=Src%2FFluentAssertions%2FExecution%2FLateBoundTestFramework.cs#L45-47

> That was a stupid idea! How are the tests going to be run without any testing framework?

Using an unsupported test framework like NSpec3 🤔

Reply:
> it says Source Not Available

You have to sign-in first (e.g. using Github OAuth)

Reply:
>  Does Coveralls merge all different test runs?

I think so. If you run `build.(sh|ps1)` you run this like in the pipeline. When finished, you get a `reports` directory beneath `Artifacts`. And there you can run the `index.html` and browse the coverage files